src/dict.js

```diff
@@ -76,6 +76,18 @@ Sk.builtin.dict.prototype.key$lookup = function (bucket, key) {
     var eq;
     var i;
 
+    // Fast path: We spend a *lot* of time looking up strings
+    // in dictionaries. (Every attribute access, for starters.)
+    if (key.ob$type === Sk.builtin.str) {
+        for (i = 0; i < bucket.items.length; i++) {
+            item = bucket.items[i];
+            if (item.lhs.ob$type === Sk.builtin.str && item.lhs.v === key.v) {
```

Comment:
I marginally prefer using ```item.lhs.$jsstr() === key.$jsstr()``` here to get away from assuming ```.v``` is the representation, but this is internal and it is so pervasive, that it probably doesn't matter...

Reply:
That function dispatch would be expensive (this code is red-hot); if there were _anywhere_ to inline `$jsstr()`, it's here.

Reply:
Fair enough.